src/grammar.rs

```diff
@@ -0,0 +1,251 @@
+use pest::prelude::*;
+
+impl_rdp! {
+    grammar! {
+        whitespace = _{ [" "]|["\t"]|["\n"]|["\r"] }
+
+        raw_text = @{ ( !["{{"] ~ any )+ }
+        raw_block_text = @{ ( !["{{{{"] ~ any )+ }
+
+// Note: this is not full and strict json literal definition, just for tokenize string,
+// array and object types which may contains whitespace. We will use a real json parser
+// for real json processing
+        literal = _{ string_literal |
+                     array_literal |
+                     object_literal |
+                     number_literal |
+                     null_literal |
+                     boolean_literal }
+
+        null_literal = @{ ["null"] }
+        boolean_literal = @{ ["true"]|["false"] }
+        number_literal = @{ ["-"]? ~ ['0'..'9']+ ~ ["."]? ~ ['0'..'9']* ~ (["E"] ~ ["-"]? ~ ['0'..'9']+)? }
+        string_literal = { ["\""] ~ (!["\""] ~ (["\\\""] | any))* ~ ["\""] }
```

Comment:
Although it doesn't make any particular difference, I would still recommend making `string_lieral` `atomic` for the sake of consistency.